pytorch_lightning/utilities/deepspeed.py

```diff
@@ -49,7 +51,7 @@
 CPU_DEVICE = torch.device("cpu")
 
 
-def ds_checkpoint_dir(checkpoint_dir: str, tag: str = None):
+def ds_checkpoint_dir(checkpoint_dir: _PATH, tag: str | None = None) -> str:
```

Comment:
n00b q: is this the new convention in favor of `Optional[str]` ? if so, should we seek to migrate these all at once to avoid having multiple conventions existing in the codebase at the same time?

Reply:
Yes, see [PEP 604](https://www.python.org/dev/peps/pep-0604/)

pre-commit automatically does the update if the file has `from __future__ import annotations`.

> should we seek to migrate these all at once

Yes. This is the plan described in #11205, but won't do it until the bug-fix releases stop since this would be a huge source of conflicts.

Since these are small files, I think it's fine to update them already, less work for later.

> avoid having multiple conventions existing in the codebase at the same time?

This is not a big problem because everything will be updated automatically later.
I personally won't mention this in code review for any new code unless the author wants to use this new notation directly.